### PR TITLE
added web dragn drop upload

### PR DESCRIPTION
**Pull request by agent-z28**
This is the web drag and drop upload module for odoo 8.0

### CODE REVIEW

File: web_drag_drop_upload/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+import controllers
```
Comment (elicoidal) [inline web_drag_drop_upload/__init__.py]: Hi @agenterp Thanks for your contribution to the OCA. 
Please follow the guidelines from:
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
I miss the README file and short headers in your py files following: https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: web_drag_drop_upload/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#
+##############################################################################
+
+{
```
Comment (elicoidal) [inline web_drag_drop_upload/__openerp__.py]: Please following template and guidelines here: 
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py
version: 8.0.1.0.0
No description (=>README)
add OCA (Odoo Community Association) in the author list
If Andreas is contributor (individual), remove from author (companies) and add him in the corresponding section of the README

File: web_drag_drop_upload/i18n/en_US.po [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline web_drag_drop_upload/i18n/en_US.po]: Remove this file

**Comment by pedrobaeza** [comment]
Thanks for this valuable contribution.

If I understand correctly the use of the module, you should rename it to _web_widget_binary_drag_and_drop_ to be in line with standard nomenclature. It's also needed to have better instructions on how to use it. Using the README template from OCA is a good start for the sections to include.

I see two different behaviours that you can extract in different modules: the widget and the sidebar drag and drop.

**Comment by agent-z28** [comment]
Hello This is all correct.

Will finish the work by tomorrow

Am 05.01.2016 um 07:42 schrieb Pedro M. Baeza:

> I see two different behaviours that you can extract in different 
> modules: the widget and the sidebar drag and drop.

## 

<#>
Agent ERP GmbH Geschäftsführender Gesellschafter _Dipl.Ing (FH) Georg 
Notter_
BALDUINSTRASSE 24 HAMBURG 20359 DEUTSCHLAND WWW.AGENT-ERP.DE <#>
- 49 171 8156666  
- 49 40 50699567  
- 49 40 30089651  

Firmensitz: Hamburg
Amtsgericht Hamburg:    HRB 128270
Ust.ID-Nr.  DE290508977
FA Hamburg  47/701/02608

Diese E-mail enthält eventuell vertrauliche und/oder rechtlich 
geschützte Informationen. Wenn Sie nicht der richtige Adressat sind oder 
diese E-mail irrtümlich erhalten haben, informieren Sie bitte sofort den 
Absender und vernichten Sie diese E-mail. Das unerlaubte Kopieren sowie 
die unerlaubte Weitergabe dieser E-mail ist nicht gestattet. This e-mail 
may contain confidential and/or privileged information. If you are not 
the intended recipient (or have received this e-mail in error) please 
notify the sender immediately and destroy this e-mail. Any unauthorised 
copying, disclosure or distribution of the material in this e-mail is 
forbidden.

**Comment by bealdav** [comment]
This module will be appreciate by everybody, I think. Thanks a lot.
My test on runbot with all modules of the branch installed display this message above header bar.

drag

thanks if you can fix it.

File: web_drag_drop_upload/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+#----------------------------------------------------------
+# OpenERP Web web Controllers
+#----------------------------------------------------------
+class Binary(http.Controller):
+
+    @http.route('/ddup/binary/upload', type='http', auth="user")
+    @serialize_exception
+    def upload(self, callback, ufile):
+        print "controller upload file"
```
Comment (hbrunn) [inline web_drag_drop_upload/controllers/main.py]: remove

File: web_drag_drop_upload/static/src/js/drag_drop_upload.js [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+    *
+    * Example:
+    * <field name="document" widget="dad_upload"/>
+    *
+    * */
+
+    instance.dad_upload = {}
+    instance.web.form.widgets.add('dad_upload', 'instance.dad_upload.DaDUploadFormWidget')
+
+    instance.dad_upload.DaDUploadFormWidget = instance.web.form.FieldBinary.extend({
```
Comment (hbrunn) [inline web_drag_drop_upload/static/src/js/drag_drop_upload.js]: this duplicates a lot of code from the above. refactor

File: web_drag_drop_upload/views/drag_drop_upload_tmpl.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates id="template" xml:space="preserve">
+<t t-name="DaDFieldBinaryFile">
```
Comment (hbrunn) [inline web_drag_drop_upload/views/drag_drop_upload_tmpl.xml]: extend `FieldBinary` and change what needs to be changed

File: web_drag_drop_upload/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    return wrap
+
+#----------------------------------------------------------
+# OpenERP Web web Controllers
+#----------------------------------------------------------
+class Binary(http.Controller):
+
+    @http.route('/ddup/binary/upload', type='http', auth="user")
+    @serialize_exception
+    def upload(self, callback, ufile):
```
Comment (hbrunn) [inline web_drag_drop_upload/controllers/main.py]: if I read the code correctly, this is an exact duplication of https://github.com/OCA/OCB/blob/8.0/addons/web/controllers/main.py#L1152 - why do you need your own controller?

File: web_drag_drop_upload/controllers/main.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            data = ufile.read()
+            args = [len(data), ufile.filename,
+                    ufile.content_type, base64.b64encode(data)]
+        except Exception, e:
+            args = [False, e.message]
+        return out % (simplejson.dumps(callback), simplejson.dumps(args))
+
+
+    @http.route('/ddup/binary/upload_attachment', type='http', auth="user")
+    def upload_attachment(self, callback, model, id, ufile):
```
Comment (hbrunn) [inline web_drag_drop_upload/controllers/main.py]: same: https://github.com/OCA/OCB/blob/8.0/addons/web/controllers/main.py#L1168

I'd propose to remove the whole file and use the core routes

Comment (agenterp) [inline web_drag_drop_upload/controllers/main.py]: okay next monday the js guy has to do it ....

File: web_drag_drop_upload/controllers/main.py (hunk 7 above)
Comment (agent-z28) [inline web_drag_drop_upload/controllers/main.py]: we had local modifications, removed it....

File: web_widget_binary_drag_and_drop/README.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============
+{Web Widget Bindary Drag and Drop}
+==============
+
+This module extends the functionality of the Sidebar to support Drag and Drop Fileuploads.
+Also multiple fileds can be uploaded at the same time.
```
Comment (bealdav) [inline web_widget_binary_drag_and_drop/README.rst]: s/fileds/files

File: web_widget_binary_drag_and_drop/README.rst [hunk 10]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============
+{Web Widget Bindary Drag and Drop}
```
Comment (bealdav) [inline web_widget_binary_drag_and_drop/README.rst]: { required ?

**Comment by pedrobaeza** [comment]
@agenterp, are you going to finish this, making Travis green?

**Comment by agent-z28** [comment]
We will provide the finished module on the OCA DAYS THIS YEAR.

**Comment by hbrunn** [comment]
I started something similar in https://github.com/OCA/web/pull/907, given the lack of activity here I'd propose to close this and backport the 10 version if needed

**Comment by pedrobaeza** [comment]
I agree.

**Pull request closed without merging**